perses/rjmc/geometry.py

```diff
@@ -318,8 +319,8 @@ def _logp_propose(self, top_proposal, old_positions, beta, new_positions=None, d
             atom_placements.append(atom_placement_array)
 
 
-            #logp_proposal += logp_r + logp_theta + logp_phi + np.log(detJ)
-            logp_proposal+=logp_r+logp_theta+logp_phi
+            logp_proposal += logp_r + logp_theta + logp_phi - np.log(detJ)
```

Comment:
Why is `log(detJ)` subtracted here instead of added? Sorry if I missed some other change that makes this obvious.

Reply:
![img_20190225_234459](https://user-images.githubusercontent.com/46943710/53924984-46247f80-404c-11e9-8ba6-0151d5883300.jpg)

Reply:
![img_1963](https://user-images.githubusercontent.com/46943710/53925497-255d2980-404e-11e9-9883-2ee252ea67a3.jpg)

Reply:
the first image is courtesy of jdc. i just rewrote his result so the -det(J) is explicit.  you can also look at the dimensionality of the result  (it must be dimensionless).  i defined P_r \propto (r^2)*exp(-k*beta(r-r0)^2/2) and same for P_theta, except instead of r^2 it is sin(theta).  i should mention that in the 
`_bond_logq` and `_angle_logq` functions, i computed the partition function numerically, which can be expensive...there are better ways of increasing the efficiency of these calculations, or perhaps they can just be omitted with a keyword

Reply:
this is great and all, but it still doesn't explain why you were sampling the wrong dihedral distributions in your simple molecule studies. that will take some more digging. but ill give it a try tomorrow

Reply:
hm, why is `J^-1` there
![image](https://user-images.githubusercontent.com/4674712/53932345-28184880-4067-11e9-8914-ca2140035b6f.png)
? Doesn't the change of variables formula involve multiplying by the absolute value of the determinant of the Jacobian matrix of the transformation (not dividing)?

Reply:
>i should mention that in the _bond_logq and _angle_logq functions, i computed the partition function numerically

I think this is fine at least for now.

Reply:
that's what i thought initially but that doesn't render the expression dimensionless (neglecting the extra 1/volume dimension from the ratio of the partition functions).  when @jchodera posted this picture on slack as a correction to a back-of-the-envelope derivation on the whiteboard from the day before, he specifically mentioned a correction to det(J).

but in any event, if we multiply by det(J) in the denominator (instead of dividing), then the dimensions don't cancel with the r^2 term and the sin(theta) term in P_r and P_theta, respectively

Reply:
I was surprised to find this too! But in working through the math, I found I needed to add J(r, theta, phi) = r^2 sin(theta) inside integrals when transforming dx dy dz into dr dtheta dphi, but the transformation of probability densities from p(x,y,z) to p(r,theta,phi) required dividing by J. See more on the transformation of probability densities here:

https://en.m.wikibooks.org/wiki/Probability/Transformation_of_Probability_Densities

Note that the probability density gets multiplied by J^-1.

This is also easy to see if you write 

p(x,y,z) dx dy dz = p(r,theta,phi) dr dtheta dphi

You can see that we should be replacing the p(x,y,z) in the denominator of our acceptable probability by something like p(r,theta,phi) * | dr dtheta dphi| / |dx dy dz| which is multiplying by J^-1.

Reply:
I was surprised to find this too! But in working through the math, I found I needed to add J(r, theta, phi) = r^2 sin(theta) inside integrals when transforming dx dy dz into dr dtheta dphi, but the transformation of probability densities from p(x,y,z) to p(r,theta,phi) required dividing by J. See more on the transformation of probability densities here:

https://en.m.wikibooks.org/wiki/Probability/Transformation_of_Probability_Densities

Note that the probability density gets multiplied by J^-1.

This is also easy to see if you write 

p(x,y,z) dx dy dz = p(r,theta,phi) dr dtheta dphi

You can see that we should be replacing the p(x,y,z) in the denominator of our acceptable probability by something like p(r,theta,phi) * | dr dtheta dphi| / |dx dy dz| which is multiplying by J^-1.

Reply:
>But in working through the math, I found I needed to add J(r, theta, phi) = r^2 sin(theta) inside integrals when transforming dx dy dz into dr dtheta dphi, but the transformation of probability densities from p(x,y,z) to p(r,theta,phi) required dividing by J.

Oh, yeah, I see what you're saying. Thanks for the clarification (and the catch!)